### PR TITLE
GitHub cicd

### PR DESCRIPTION
**Pull request by HelgaStock**
added GitHub workflow ci/cd pipeline

### CODE REVIEW

File: architecture/.github\workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          terraform_version: 1.1.7
+          sh upload-src.sh
+      - name: deploy infrastructure
+        run: |
+          cd AWS-SECURITYDEMO
+          sh setup_s3.sh
+          terraform init
+          terraform apply -auto-approve
+
```
Comment (fabianschmauder) [inline architecture/.github\workflows/deploy.yml:28]: missing line

File: architecture/.github\workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_SESSION_TOKEN: ${{ secrets.AWS_SESSION_TOKEN }}
+      AWS_REGION: us-west-2
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.1.7
+          sh upload-src.sh
```
Comment (fabianschmauder) [inline architecture/.github\workflows/deploy.yml:21]: this line should not be in the terraform action. 
add this line to deploy infrastructure or add a new step like build

File: architecture/.github\workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    env:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_SESSION_TOKEN: ${{ secrets.AWS_SESSION_TOKEN }}
+      AWS_REGION: us-west-2
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.1.7
```
Comment (fabianschmauder) [inline architecture/.github\workflows/deploy.yml:20]: you can also remove the with or update to a new version.
This is already an old version.

Could be that I used this. Sorry for that

**Review by fabianschmauder — COMMENTED** [review]
Just a few changes on the pipeline

**Review by HelgaStock — COMMENTED** [review]
all changes implemented

**Pull request merged**